File: Python/queen.py
def queen(n):
    ans = [0]
    upperlim = (1 << n) - 1
    def test(row, ld, rd):
        p = 0
        pos = 0
        if row != upperlim:
            pos = upperlim & (~(row | ld | rd ))
            while pos != 0:
                p = pos & (~pos + 1)
                pos = pos - p
                test(row | p, (ld | p) << 1, (rd | p) >> 1);  
        else:
            ans[0] += 1
    test(0, 0, 0)
    return ans[0]

print(queen(8))

File: C/queen.c
#include <stdio.h>
int main() {
    printf("%d\n", queen(4));
    return 0;
}
int queen(int n) {
    return queen(0, 0, 0, 0)
}
int _queen(int l, int r, int m, int k) {
    int ans = 0;

    for(int i = (~(l | r | m)) & 0xff; i; i -= i & -i) {
        ans += queen((l | (i & -i)) << 1, (r | (i & -i)) >> 1 , m | (i & -i), k + 1);
    }

    return k == 8 ? 1 : ans;
}
